### PR TITLE
KOGITO-329 - GraphQL subscription for Process, User Tasks and domain objects

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by mswiderski** [comment]
is there any example how to make use of it? I'd like to try this out with travel agency app so see it in action...

**Comment by cristianonicolai** [comment]
@mswiderski the only client now is the one used in the tests (websocket api). In the next steps, I will setup the GraphiQL UI and Travel agency with a GraphQL JS client to consume these events, so that work will drive us to create a sample configuration on how to actually consume the subscription events. Ajay is already working on that.

File: data-index/data-index-service/src/test/java/org/kie/kogito/index/graphql/WebSocketSubscriptionTest.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+    @Inject
+    ReactiveMessagingEventConsumer consumer;
+
+    @Inject
+    ProtobufService protobufService;
+
+    @Inject
+    Vertx vertx;
+
+//    @Test
```
Comment (nmirasch) [inline data-index/data-index-service/src/test/java/org/kie/kogito/index/graphql/WebSocketSubscriptionTest.java]: @cristianonicolai this commented code is the one that uses assertProcessInstanceSubscription and assertUserTaskInstanceSubscription, is this intentionally or you missed to uncomment?

Comment (cristianonicolai) [reply data-index/data-index-service/src/test/java/org/kie/kogito/index/graphql/WebSocketSubscriptionTest.java]: @nmirasch sorry, please wait a bit until I can get the test to pass, I'm playing with the code to find out why it only fails on Jenkins :/

File: data-index/data-index-storage/data-index-storage-infinispan/src/main/java/org/kie/kogito/index/infinispan/listener/CacheObjectRemovedListener.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.function.Consumer;
+
+import org.infinispan.client.hotrod.annotation.ClientCacheEntryRemoved;
+import org.infinispan.client.hotrod.annotation.ClientListener;
+import org.infinispan.client.hotrod.event.ClientCacheEntryRemovedEvent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@ClientListener
+public class CacheObjectRemovedListener<K> {
```
Comment (nmirasch) [inline data-index/data-index-storage/data-index-storage-infinispan/src/main/java/org/kie/kogito/index/infinispan/listener/CacheObjectRemovedListener.java]: @cristianonicolai  Does this  class have to extend from AbstractCacheObjectListener like create or update, I mean  using this handleEvent(K key, long version) or Is the CacheObjectRemove a different case?

Comment (nmirasch) [reply data-index/data-index-storage/data-index-storage-infinispan/src/main/java/org/kie/kogito/index/infinispan/listener/CacheObjectRemovedListener.java]: @cristianonicolai sorry I saw that ClientCacheEntryRemovedEvent doesn't come with version :)

File: data-index/data-index-storage/data-index-storage-infinispan/src/main/java/org/kie/kogito/index/infinispan/listener/CacheObjectRemovedListener.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    private Consumer<K> consumer;
+
+    public CacheObjectRemovedListener(Consumer<K> consumer) {
+        this.consumer = consumer;
+    }
+
+    @ClientCacheEntryRemoved
+    public void handleRemovedEvent(ClientCacheEntryRemovedEvent<K> e) {
+        LOGGER.debug("Handle create event for entry with id: {}", e.getKey());
```
Comment (nmirasch) [inline data-index/data-index-storage/data-index-storage-infinispan/src/main/java/org/kie/kogito/index/infinispan/listener/CacheObjectRemovedListener.java]: @cristianonicolai maybe this debug message have to say 'remove event'?

Comment (cristianonicolai) [reply data-index/data-index-storage/data-index-storage-infinispan/src/main/java/org/kie/kogito/index/infinispan/listener/CacheObjectRemovedListener.java]: true, fixed now.

File: data-index/data-index-storage/data-index-storage-infinispan/src/main/java/org/kie/kogito/index/infinispan/listener/CacheObjectRemovedListener.java (hunk 2 above)
Comment (cristianonicolai) [reply data-index/data-index-storage/data-index-storage-infinispan/src/main/java/org/kie/kogito/index/infinispan/listener/CacheObjectRemovedListener.java]: yep, that's why. :)

**Comment by cristianonicolai** [comment]
@mswiderski @nmirasch ready for review again, thanks.

**Pull request merged**